File: threedollar-api/src/main/java/com/depromeet/threedollar/api/controller/TestController.java
package com.depromeet.threedollar.api.controller;

import com.depromeet.threedollar.api.common.dto.ApiResponse;
import com.depromeet.threedollar.api.config.session.UserSession;
import com.depromeet.threedollar.api.service.auth.dto.response.LoginResponse;
import com.depromeet.threedollar.domain.domain.user.User;
import com.depromeet.threedollar.domain.domain.user.UserRepository;
import com.depromeet.threedollar.domain.domain.user.UserSocialType;
import io.swagger.annotations.ApiOperation;
import lombok.RequiredArgsConstructor;
import org.springframework.context.annotation.Profile;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpSession;

import static com.depromeet.threedollar.api.config.session.SessionConstants.USER_SESSION;

@Profile({"local", "local-will", "dev"})
@RequiredArgsConstructor
@RestController
public class TestController {

    private static final User testUser = User.newInstance("test-uid", UserSocialType.KAKAO, "관리자 계정");

    private final HttpSession httpSession;
    private final UserRepository userRepository;

    @ApiOperation("[테스트용] 테스트를 위한 토큰을 받아옵니다.")
    @GetMapping("/test-token")
    public ApiResponse<LoginResponse> getTestToken() {
        User user = userRepository.findUserBySocialIdAndSocialType(testUser.getSocialId(), testUser.getSocialType());
        if (user == null) {
            user = userRepository.save(testUser);
        }
        httpSession.setAttribute(USER_SESSION, UserSession.of(user.getId()));
        return ApiResponse.success(LoginResponse.of(httpSession.getId()));
    }

}

File: threedollar-admin/src/main/kotlin/com/depromeet/threedollar/admin/controller/TestController.kt
package com.depromeet.threedollar.admin.controller

import com.depromeet.threedollar.admin.common.dto.ApiResponse
import com.depromeet.threedollar.admin.service.token.TokenService
import com.depromeet.threedollar.admin.service.token.dto.AdminTokenDto
import com.depromeet.threedollar.domain.domain.admin.AdminCreator
import com.depromeet.threedollar.domain.domain.admin.AdminRepository
import org.springframework.context.annotation.Profile
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@Profile("local")
@RestController
class TestController(
    private val tokenService: TokenService,
    private val adminRepository: AdminRepository
) {

    companion object {
        private const val TEST_ADMIN_EMAIL = "test.admin@test.com"
        private const val TEST_ADMIN_NAME = "테스트 관리자"
    }

    @GetMapping("/test-token")
    fun getTestToken(): ApiResponse<String> {
        val admin = adminRepository.findAdminByEmail(TEST_ADMIN_EMAIL)
            ?: adminRepository.save(AdminCreator.create(TEST_ADMIN_EMAIL, TEST_ADMIN_NAME))
        return ApiResponse.success(tokenService.encode(AdminTokenDto(admin.id)))
    }

}
